prov/shm/src/smr.h

```diff
@@ -150,21 +139,22 @@ struct smr_pend_entry {
 	ofi_hmem_async_event_t	async_event;
 };
 
-struct smr_match_attr {
-	fi_addr_t	id;
-	uint64_t	tag;
-	uint64_t	ignore;
-};
-
-static inline int smr_match_id(fi_addr_t addr, fi_addr_t match_addr)
+static inline enum fi_hmem_iface smr_get_mr_hmem_iface(struct util_domain *domain,
+				void **desc, uint64_t *device)
 {
-	return (addr == FI_ADDR_UNSPEC) || (match_addr == FI_ADDR_UNSPEC) ||
-		(addr == match_addr);
+	if (!(domain->mr_mode & FI_MR_HMEM) || !desc || !*desc) {
+		*device = 0;
+		return FI_HMEM_SYSTEM;
+	}
+
+	*device = ((struct ofi_mr *) *desc)->device;
+	return ((struct ofi_mr *) *desc)->iface;
 }
 
-static inline int smr_match_tag(uint64_t tag, uint64_t ignore, uint64_t match_tag)
+static inline uint64_t smr_get_mr_flags(void **desc)
 {
-	return ((tag | ignore) == (match_tag | ignore));
+	assert(desc && *desc);
+	return ((struct ofi_mr *) *desc)->flags;
 }
```

Comment:
Where are they used?

Reply:
I think this is a weird git-ism. This patch didn't touch this function, just deleted the matching functions around it and git got confused. But you're right this isn't needed anymore. It was left unused when we moved to passing the ofi_mr around instead of the desc to support the multiple iface iovs. It was used in what now looks like this:
```
		use_ipc = ofi_hmem_is_ipc_enabled(iface) &&
				smr_desc->flags & FI_HMEM_DEVICE_ONLY &&
				!(op_flags & FI_INJECT);
```
I will add a patch to remove it. Thanks!

Reply:
Ha nevermind. That was a me-ism from not resolving a conflict correctly after rebasing 😅